gnovm/pkg/gnolang/values_conversions.go

```diff
@@ -847,156 +847,155 @@ GNO_CASE:
 		switch k {
 		case IntKind:
 			validate(Float32Kind, IntKind, func() bool {
-				val := float64(tv.GetFloat32())
-				trunc := math.Trunc(val)
+				trunc := tv.GetFloat32().Trunc()
 
-				if val != trunc {
+				if !trunc.Eq(tv.GetFloat32()) {
 					return false
 				}
 
-				return int64(trunc) >= math.MinInt && int64(trunc) <= math.MaxInt
+				truncInt64 := trunc.Int64()
+				return truncInt64 >= math.MinInt && truncInt64 <= math.MaxInt
 			})
 
-			x := int(tv.GetFloat32()) // XXX determinism?
+			x := tv.GetFloat32().Int()
 			tv.T = t
 			tv.SetInt(x)
 		case Int8Kind:
 			validate(Float32Kind, Int8Kind, func() bool {
-				val := float64(tv.GetFloat32())
-				trunc := math.Trunc(val)
+				trunc := tv.GetFloat32().Trunc()
```

Comment:
Can we not use something like F32toint32 here, and all other places?

This would also avoid us having to use Trunc and Mod

Reply:
Don't know how you will remove the float part without a trunc ? Maybe you think about a new approach ?

Reply:
apologies, I see now this is only used for constants.

Looks good then